simplesat/dependency_solver.py

```diff
@@ -65,12 +71,62 @@ def _create_rules(self, request):
 
         rules_generator = RulesGenerator(pool, request, installed_map)
 
-        return list(rules_generator.iter_rules())
+        return requirement_ids, list(rules_generator.iter_rules())
+
+
+def _connected_packages(solution, requirement_ids, pool):
+    """ Return packages which are associated with a requirement. """
```

Comment:
can we call `requirement_ids` `pkg_ids` instead ? IIUC, it does not matter for this function that packages are coming from requirements or not.